### PR TITLE
Fix item query not found exceptions

### LINKED ISSUE #256: Fix not found exceptions
**Describe the bug**
If the container does not exists and a query is done it will throw a not found exception. This should return a response with status code.

delete the container if it exists
create the container and insert a item
query the item (SELECT * from root WHERE true)
delete the container
throws (SELECT * from root WHERE true)

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description
The caches still throw exceptions which causes the stream API to throw. 

Actual behavior:
```
await container.DeleteAsync();

var queryIterator = container.Items.CreateItemQueryAsStream("select * from t where true", maxConcurrency: 1, partitionKey: "testpk");

//Throws not found exception
var response = await queryIterator.FetchNextSetAsync());

```
Expected behavior:
```
await container.DeleteAsync();

var queryIterator = container.Items.CreateItemQueryAsStream("select * from t where true", maxConcurrency: 1, partitionKey: "testpk");

var response = await queryIterator.FetchNextSetAsync());
Assert.IsNotNull(response);
Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
```

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues

closes #256

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Item/CosmosItemsCore.cs [hunk 1]
```diff
@@ -525,8 +526,61 @@ internal class CosmosItemsCore : CosmosItems
...
+                    responseHeaders: CosmosQueryResponseMessageHeaders.ConvertToQueryHeaders(exception.Headers),
+                    requestMessage: null,
+                    errorMessage: exception.Message,
+                    statusCode: exception.StatusCode,
+                    error: exception.Error);
+            }
+            catch (AggregateException ae)
+            {
+                AggregateException innerExceptions = ae.Flatten();
+                Exception exception = innerExceptions.InnerExceptions.FirstOrDefault(innerEx => innerEx is CosmosException);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Item/CosmosItemsCore.cs]: TryGet helper that returns CosmosException and DCE as out parameters? Resused in TransportHandler? (copying this comment from the other PR)

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Item/CosmosItemsCore.cs]: Done

File: Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs [hunk 2]
```diff
@@ -88,6 +87,30 @@ public TransportHandler(CosmosClient client)
             }
         }
 
+        internal static CosmosResponseMessage AggregateExceptionConverter(AggregateException aggregateException, CosmosRequestMessage request)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs]: Thanks u.

File: Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs [hunk 3]
```diff
@@ -48,19 +48,18 @@ public TransportHandler(CosmosClient client)
                 // TODO: because the SDK underneath this path uses ContinueWith or task.Result we need to catch AggregateExceptions here
                 // in order to ensure that underlying DocumentClientExceptions get propagated up correctly. Once all ContinueWith and .Result 
                 // is removed this catch can be safely removed.
-                AggregateException innerExceptions = ex.Flatten();
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Handler/TransportHandler.cs]: Please add UT capturing the contract handling for DCE, CsosmoException and others.

File: Microsoft.Azure.Cosmos/src/Resource/Item/CosmosItemsCore.cs [hunk 4]
```diff
@@ -525,8 +527,35 @@ internal class CosmosItemsCore : CosmosItems
...
         {
-            CosmosQueryExecutionContext cosmosQueryExecution = (CosmosQueryExecutionContext)state;
-            return (CosmosResponseMessage)(await cosmosQueryExecution.ExecuteNextAsync(cancellationToken));
+            // This catches exception thrown by the caches and converts it to QueryResponse
+            try
+            {
+                CosmosQueryExecutionContext cosmosQueryExecution = (CosmosQueryExecutionContext)state;
+                return (CosmosResponseMessage)(await cosmosQueryExecution.ExecuteNextAsync(cancellationToken));
+            }
+            catch (DocumentClientException exception)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Item/CosmosItemsCore.cs]: Thoughts on avoiding this logic spill across?

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Item/CosmosItemsCore.cs]: I created an [issue](https://github.com/Azure/azure-cosmos-dotnet-v3/issues/275) that will fix this. I think our main problem is the caches are still throwing, but we shouldn't be for all the stream apis.

**Pull request merged**